pkg/phlaredb/schemas/v1/locations.go

```diff
@@ -27,15 +23,53 @@ func (*LocationPersister) SortingColumns() parquet.SortingOption {
 	)
 }
 
-func (*LocationPersister) Deconstruct(row parquet.Row, id uint64, l *profilev1.Location) parquet.Row {
+func (*LocationPersister) Deconstruct(row parquet.Row, _ uint64, l *InMemoryLocation) parquet.Row {
 	row = locationsSchema.Deconstruct(row, l)
 	return row
 }
 
-func (*LocationPersister) Reconstruct(row parquet.Row) (id uint64, l *profilev1.Location, err error) {
-	var location profilev1.Location
+func (*LocationPersister) Reconstruct(row parquet.Row) (uint64, *InMemoryLocation, error) {
+	var location InMemoryLocation
 	if err := locationsSchema.Reconstruct(&location, row); err != nil {
 		return 0, nil, err
 	}
 	return 0, &location, nil
 }
+
+type InMemoryLocation struct {
+	// Unique nonzero id for the location.  A profile could use
+	// instruction addresses or any integer sequence as ids.
+	Id uint64
+	// The instruction address for this location, if available.  It
+	// should be within [Mapping.memory_start...Mapping.memory_limit]
+	// for the corresponding mapping. A non-leaf address may be in the
+	// middle of a call instruction. It is up to display tools to find
+	// the beginning of the instruction if necessary.
+	Address uint64
+	// The id of the corresponding profile.Mapping for this location.
+	// It can be unset if the mapping is unknown or not applicable for
+	// this profile type.
+	MappingId uint32
+	// Provides an indication that multiple symbols map to this location's
+	// address, for example due to identical code folding by the linker. In that
+	// case the line information above represents one of the multiple
+	// symbols. This field must be recomputed when the symbolization state of the
+	// profile changes.
+	IsFolded bool
```

Comment:
struct padding should be improved

Reply:
There are two changes comparing to the proto definition:
1. `MappingId` type uint64 -> uint32;
2. `IsFolded` moved next to `MappingId`, to avoid extra-padding.

The current type size is 48 bytes, given the 8-bytes alignment. Further improvement does not seem to be possible.